charon/src/ast/values.rs

```diff
@@ -81,3 +82,30 @@ pub enum ScalarValue {
     U64(u64),
     U128(u128),
 }
+
+/// This is simlar to the Scalar value above. However, instead of storing
+/// the float value itself, we store its bits representation using a u128.
+/// This allows to derive the Eq and Ord traits, which are not implemented
+/// for floats
+#[derive(
+    Debug,
+    PartialEq,
+    Eq,
+    Copy,
+    Clone,
+    EnumIsA,
+    EnumAsGetters,
+    VariantName,
+    VariantIndexArity,
+    Hash,
+    PartialOrd,
+    Ord,
+    Drive,
+    DriveMut,
+)]
+pub enum FloatValue {
+    F16(u128),
+    F32(u128),
+    F64(u128),
+    F128(u128),
+}
```

Comment:
I think it's worth adding a comment explaining why we store the float values as `u128`

Reply:
As discussed in person, the comment (hidden behind the stack of Derives) is fine